docs/designs/upgrade-design/upgrade-design-doc.md

```diff
@@ -0,0 +1,653 @@
+- **PostgreSQL configuration**: Streaming replication with synchronous commit enabled (for standby server synchronization)
+
+**Configuration Examples**: See [CNPG HA Configuration Examples](./commands.md#cnpg-ha-configuration-examples) for complete YAML specifications.
+
+#### Zero-Downtime Upgrade Sequence
+
+**CNPG Managed Rolling Update Process:**
+
+1. **Standby Server Upgrade Phase** (Automatic):
+   - CNPG upgrades standby instances first (highest serial number to lowest)
+   - Each standby server downloads new images and restarts with new configuration
+   - Primary continues serving traffic with full availability
+
+2. **Controlled Switchover Phase** (Manual with Supervised Mode):
+   - Check standby lag before switchover to ensure optimal timing
+   - Manual switchover to most aligned standby server using CNPG promote command
+   - Service endpoints automatically update to new primary
+
+3. **Primary Upgrade Phase** (Automatic):
+   - Former primary becomes standby server and receives upgrade
+   - New primary (former standby server) continues serving traffic
+   - Zero service interruption during transition
+
+**Command Examples**: See [CNPG Zero-Downtime Upgrade Sequence](./commands.md#cnpg-zero-downtime-upgrade-sequence) for step-by-step commands.
+
+#### Operational Benefits of CNPG HA
+
+**✅ Built-in Capabilities:**
+- **WAL-based replication**: Ensures data consistency during failover between primary and standby servers
+- **Automatic endpoint management**: DNS and service updates during switchover
+- **Connection draining**: Graceful client connection handling
+- **Monitoring integration**: Real-time standby lag and health metrics
+- **Rollback support**: Can revert primary assignment if issues detected
+
+**⚠️ Operational Considerations:**
+- **Supervised mode**: Production environments use manual switchover for maximum control
+- **Standby lag monitoring**: Check lag before manual switchover for optimal timing
+- **Connection pool awareness**: Applications should use read/write service endpoints for seamless failover
+- **Monitoring integration**: Real-time health checks help inform manual switchover decisions
+
+
+#### Risk Mitigation Enhancements
+
+**Enhanced with CNPG Specifics:**
+- **Automatic failover**: Unplanned failures trigger immediate CNPG failover (`.spec.failoverDelay: "0s"`)
+- **Planned maintenance**: Supervised upgrades allow optimal timing for switchover
+- **Data protection**: WAL streaming and synchronous replication prevent data loss between primary and standby servers
+- **Service continuity**: Kubernetes service endpoints automatically update during failover
+- **Monitoring integration**: CNPG status commands provide real-time cluster health visibility
+
+This CNPG-based HA strategy ensures DocumentDB clusters achieve true zero-downtime upgrades while maintaining data integrity and operational simplicity.
+
+## Multi-Node Upgrade Strategy (Future Enhancement)
```

Comment:
Multi node and remote regions are two orthogonal concepts so pleae treat them as such